packages/react-components/src/components/ChatMessage/ChatMessageContent.tsx

```diff
@@ -78,16 +80,18 @@ const MessageContentWithLiveAria = (props: MessageContentWithLiveAriaProps): JSX
 const MessageContentAsRichTextHTML = (props: ChatMessageContentProps): JSX.Element => {
   /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
   useEffect(() => {
-    props.message.attachedFilesMetadata?.map((fileMetadata) => {
-      if (
-        props.onFetchAttachment &&
-        props.attachmentsMap &&
-        fileMetadata.attachmentType === 'inlineImage' &&
-        props.attachmentsMap[fileMetadata.id] === undefined
-      ) {
-        props.onFetchAttachment(fileMetadata);
-      }
+    const attachments = props.message.attachedFilesMetadata?.filter((fileMetadata) => {
+      return fileMetadata.attachmentType === 'inlineImage';
     });
+
+    if (props.attachmentsMap && attachments) {
+      attachments.forEach((fileMetadata) => {
+        if (props.onFetchAttachment && props.attachmentsMap && props.attachmentsMap[fileMetadata.id] === undefined) {
```

Comment:
We might be able to remove the check for props.attachmentsMap since it's been checked above.

Reply:
This throw a linter error (and honestly I am not sure why) about the attachmentsMap being undefined. So I will need to leave this condition in there